packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -612,7 +614,7 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
     >
       <FocusRects />
       <div
-        role="grid"
+        role={role}
```

Comment:
Note that even if you customize prop, the `DetailsRow` element all supply `role='row'` unless you also supply `onRenderRow`. Consider making it so the default `onRenderRow` omits `role` if `this.props.role` is overridden so that it doesn't produce an invalid semantic. This is what `List` does too if a custom `role` is supplied.

Reply:
Made changes as per discussion

Reply:
This change needs to probably happen both in `onRenderRow` on line `874` in this file and on `DetailsRow.base.tsx` to correctly account for this.

Reply:
Made the changes